src/nix-find-roots/lib/find-roots.cc

```diff
@@ -0,0 +1,233 @@
+/**
+ * @file
+ *
+ * A very simple utility to trace all the gc roots through the file-system
+ * The reason for this program is that tracing these roots is the only part of
+ * Nix that requires to run as root (because it requires reading through the
+ * user home directories to resolve the indirect roots)
+ *
+ * This program intentionnally doesnt depend on any Nix library to reduce the attack surface.
+ */
+
+#include <regex>
+#include <unistd.h>
+#include <vector>
+#include <algorithm>
+#include <fstream>
+#include <optional>
+
+#include "find-roots.hh"
+
+
+namespace nix::roots_tracer {
+namespace fs = std::filesystem;
+using std::set, std::string;
```

Comment:
```suggestion
```
Just use `std::string` etc. We got rid of the `string` alias a while ago (df552ff53e68dff8ca360adbdbea214ece1d08ee).